aiodns/__init__.py

```diff
@@ -70,28 +65,28 @@ def __init__(self, nameservers: Optional[Sequence[str]] = None,
                                         **kwargs)
         if nameservers:
             self.nameservers = nameservers
-        self._read_fds = set() # type: Set[int]
-        self._write_fds = set() # type: Set[int]
-        self._timer = None  # type: Optional[asyncio.TimerHandle]
+        self._read_fds: set[int] = set()
+        self._write_fds: set[int] = set()
+        self._timer: Optional[asyncio.TimerHandle] = None
 
     @property
     def nameservers(self) -> Sequence[str]:
         return self._channel.servers
 
     @nameservers.setter
-    def nameservers(self, value: Sequence[str]) -> None:
+    def nameservers(self, value: Iterable[Union[str, bytes]]) -> None:
         self._channel.servers = value
 
     @staticmethod
-    def _callback(fut: asyncio.Future, result: Any, errorno: int) -> None:
+    def _callback(fut: asyncio.Future[_T], result: _T, errorno: Optional[int]) -> None:
         if fut.cancelled():
             return
         if errorno is not None:
             fut.set_exception(error.DNSError(errorno, pycares.errno.strerror(errorno)))
         else:
             fut.set_result(result)
 
-    def query(self, host: str, qtype: str, qclass: Optional[str]=None) -> asyncio.Future:
+    def query(self, host: str, qtype: str, qclass: Optional[str]=None) -> asyncio.Future[Any]:
```

Comment:
All the `asyncio.Future[Any]`s are because I was struggling to understand the actual return types from pycares. Would be good if someone can help figure that out.

Reply:
@saghul Not sure if you'd like to help here. I can add the annotations myself if you can point me to the correct types, or feel free to edit the PR directly.

Reply:
I can return any of these types, depends on the query type: https://github.com/saghul/pycares/blob/90bd7fc12910d53c9982d0dd0a0f658683487212/src/pycares/__init__.py#L635 all the ones with `query` in the name.

Reply:
Looks alright to me!